android/src/main/java/software/solid/fluttervlcplayer/FlutterVlcPlayerPlugin.java

```diff
@@ -1,17 +1,41 @@
 package software.solid.fluttervlcplayer;
 
+import androidx.annotation.NonNull;
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.plugin.common.BinaryMessenger;
 import io.flutter.plugin.common.PluginRegistry.Registrar;
 
 /** FlutterVlcPlayerPlugin */
-public class FlutterVlcPlayerPlugin {
+public class FlutterVlcPlayerPlugin implements FlutterPlugin {
+
+  public FlutterVlcPlayerPlugin() {}
 
   /**
    * Plugin registration.
    */
   public static void registerWith(Registrar registrar) {
-    registrar.platformViewRegistry().registerViewFactory(
-            "flutter_video_plugin/getVideoView",
-            new FlutterVideoViewFactory(registrar.messenger(), registrar)
-    );
+    registrar
+            .platformViewRegistry()
+            .registerViewFactory(
+                    "plugins.flutter.io/webview",
```

Comment:
this should be "flutter_video_plugin/getVideoView".

Reply:
I tested all the latest changes you commits you have made in the android files. 
release 4.0.1 is lags and loses state on orientation change or widget size changes.
comparing it with yours, your code make video screen black after orientation change.

Reply:
That's because there is no runnable closure.